pkg/targets/adapter/azureservicebustarget/adapter.go

```diff
@@ -0,0 +1,271 @@
+		Namespace:     envAcc.GetNamespace(),
+		Name:          envAcc.GetName(),
+	}
+
+	metrics.MustRegisterEventProcessingStatsView()
+
+	env := envAcc.(*envAccessor)
+
+	replier, err := targetce.New(env.Component, logger.Named("replier"),
+		targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+		targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeAzureServiceBusGenericResponse),
+		targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(env.CloudEventPayloadPolicy)))
+	if err != nil {
+		logger.Panicf("Error creating CloudEvents replier: %v", err)
+	}
+
+	entityID, err := parseServiceBusResourceID(env.EntityResourceID)
+	if err != nil {
+		logger.Panicw("Unable to parse entity ID "+strconv.Quote(env.EntityResourceID), zap.Error(err))
+	}
+
+	client, err := clientFromEnvironment(entityID, newAzureServiceBusClientOptions(
+		webSocketsClientOption(env.WebSocketsEnable)))
+	if err != nil {
+		logger.Panicw("Unable to obtain interface for Service Bus Namespace", zap.Error(err))
+	}
+
+	sender, err := client.NewSender(entityID.ResourceName, nil)
+	if err != nil {
+		logger.Panicw("Unable to obtain sender for Service Bus Namespace", zap.Error(err))
+	}
+
+	return &adapter{
+		sender: sender,
+
+		discardCEContext: env.DiscardCEContext,
+		replier:          replier,
+		ceClient:         ceClient,
+		logger:           logger,
+
+		sr: metrics.MustNewEventProcessingStatsReporter(mt),
+	}
+}
+
+var _ pkgadapter.Adapter = (*adapter)(nil)
+
+type adapter struct {
+	sender *azservicebus.Sender
+
+	discardCEContext bool
+	replier          *targetce.Replier
+	ceClient         cloudevents.Client
+	logger           *zap.SugaredLogger
+
+	sr *metrics.EventProcessingStatsReporter
+}
+
+// Returns if stopCh is closed or Send() returns an error.
+func (a *adapter) Start(ctx context.Context) error {
+	a.logger.Info("Starting Azure Service Bus Target adapter")
+	return a.ceClient.StartReceiver(ctx, a.dispatch)
+}
+
+func (a *adapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
+	ceTypeTag := metrics.TagEventType(event.Type())
+	ceSrcTag := metrics.TagEventSource(event.Source())
+
+	start := time.Now()
+	defer func() {
+		a.sr.ReportProcessingLatency(time.Since(start), ceTypeTag, ceSrcTag)
+	}()
+
+	var msg []byte
+	if a.discardCEContext {
+		msg = event.Data()
+	} else {
+		jsonEvent, err := json.Marshal(event)
+		if err != nil {
+			a.logger.Errorw("Error marshalling CloudEvent", zap.Error(err))
+			a.sr.ReportProcessingError(true, ceTypeTag, ceSrcTag)
+			return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
+		}
+		msg = jsonEvent
+	}
+	err := a.sender.SendMessage(ctx, &azservicebus.Message{
+		Body: msg,
+	}, nil)
+	if err != nil {
+		a.logger.Errorw("Error sending message to Service Bus", zap.Error(err))
+		a.sr.ReportProcessingError(true, ceTypeTag, ceSrcTag)
+		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
+	}
```

Comment:
nit: could be slightly more concise
```suggestion
	if err := a.sender.SendMessage(ctx, &azservicebus.Message{Body: msg}, nil);  err != nil {
		a.logger.Errorw("Error sending message to Service Bus", zap.Error(err))
		a.sr.ReportProcessingError(true, ceTypeTag, ceSrcTag)
		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
	}
```